docs/contribute/module.md

```diff
@@ -11,18 +12,89 @@ Each experimentalist, experiment runner or theorist is a "child" package based o
     However, if your contribution requires additional dependencies, you can submit it as a full package following 
     this guide. 
 
+
+## Implementing your module
+
+After setting up your repository and linking it to your GitHub account, you can start implementing your module.
+
+### Step 1: Implement Your Code
+
+You may implement your code in the ``init.py`` located in the respective feature folder in ``src/autora``.
+
+Please refer to the following guides on implementing
+- [theorists](theorist-module.md)
+- [experimentalists](experimentalist-module.md)
+- [experiment runners](experiment-runner-module.md)
+
+If the feature you seek to implement does not fit in any of these categories, then 
+you can you can create folders for new categories. If you are unsure how to proceed, you are always welcome 
+to ask for help in the [AutoRA forum](https://github.com/orgs/AutoResearch/discussions/categories/module-contributions).
+
+### Step 2 (Optional): Add Tests
+
+It is highly encouraged to add unit tests to ensure your code is working as intended. These can be [doctests](https://docs.python.org/3/library/doctest.html) or as test cases in `tests/test_your_contribution_name.py`.
+For example, if you are implementing an experiment sampler, you may rename and modify the 
+``tests/test_experimentalist_sampler_example.py``.
+
+*Note: Tests are required if you wish that your feature becomes part of the main 
+[autora](https://github.com/AutoResearch/autora) package. However, regardless of whether you choose to implement tests, 
+you will still be able to install your package separately, in addition to autora.* 
+
+### Step 3 (Optional): Add Documentation
+
+It is highly encouraged that you add documentation of your package in your `docs/index.md`. You can also add new pages 
+in the `docs` folder. Update the `mkdocs.yml` file to reflect structure of the documentation. For example, you can add 
+new pages or delete pages that you deleted from the `docs` folder.
+
+You are also encouraged to describe basic usage of your theorist in the 
+python notebook ``Basic Usage.ipynb`` in the `docs` folder. Finally you can outline the basic setup of your theorist in 
+the `docs/quickstart.md` file.
+
+*Note: Docmentation is required if you wish that your feature becomes part of the main 
+[autora](https://github.com/AutoResearch/autora) package. However, regardless of whether you choose to write
+documentation, you will still be able to install your package separately, in addition to autora.*
+
+### Step 4: Add Dependencies
+
+In pyproject.toml add the new dependencies under `dependencies`
+
+Install the added dependencies
+```shell
+pip install -e ".[dev]"
+```
+
+## Publishing your module
+
+There are several ways to publish your package, depending on how you set up your repository.
+
+- If you used the **cookiecutter template**, and uploaded your repository to 
+github.com, then you can use Github Actions to automatically publish your package to PyPI. 
```

Comment:
```suggestion
github.com, then you can use Github Actions to automatically publish your package to PyPI or Conda. 
```

Reply:
per comment above, not sure which we want to include discussion of Conda, but if we do, we should be consistent.